### PR TITLE
Adjust timestamp based on precision

### PR DESCRIPTION
**Pull request by sambaiz**
Format using adjusted timestamp based on time_precision.

### CODE REVIEW

File: fluent-plugin-influxdb.gemspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ Gem::Specification.new do |s|
   s.test_files    = s.files.grep(%r{^(test|spec|features)/})
   s.require_paths = ["lib"]
 
-  s.add_runtime_dependency "fluentd", [">= 0.10.49", "< 2"]
+  s.add_runtime_dependency "fluentd", [">= 0.10.49", "< 0.14"]
```
Comment (repeatedly) [inline fluent-plugin-influxdb.gemspec]: Why reject v0.14 or later version?

File: lib/fluent/plugin/out_influxdb.rb [hunk 2]
```diff
@@ -163,4 +163,26 @@ def write(chunk)
       end
     end
   end
+
+  def precision_time(time)
+    # nsec is supported from v0.14
+    nstime = time * (10 ** 9) + (defined?(Fluent::EventTime) ? time.nsec : 0)
+    case @time_precision
+    when "h" then
```
Comment (repeatedly) [inline lib/fluent/plugin/out_influxdb.rb]: Using symbol is faster than string.

File: lib/fluent/plugin/out_influxdb.rb [hunk 3]
```diff
@@ -163,4 +163,26 @@ def write(chunk)
...
+    when "s" then
+      nstime / (10 ** 9)
+    when "ms" then
+      nstime / (10 ** 6)
+    when "u" then
+      nstime / (10 ** 3)
+    when "ns" then
+      nstime
+    else
+      raise Fluent::ConfigError, 'time_precision ' + @time_precision + ' is invalid.' +
```
Comment (repeatedly) [inline lib/fluent/plugin/out_influxdb.rb]: ConfigError check should be done in configure, not plugin running phase.

File: lib/fluent/plugin/out_influxdb.rb (hunk 2 above)
Comment (repeatedly) [inline lib/fluent/plugin/out_influxdb.rb]: BTW, set proc in configure and call it in precision_time is more better.

**Comment by sambaiz** [comment]
Thanks a lot! I fixed.

**Pull request merged**